pints/_mcmc/_slice_stepout.py

```diff
@@ -0,0 +1,430 @@
+
+        # Now that we have expanded the interval, set flag
+        self._interval_found = True
+
+        # Sample new trial point by sampling uniformly from the interval ``I=(l,r)``
+        self._u = np.random.uniform()
+        self._proposed[self._i] = self._l + self._u*(self._r - self._l)
+
+        # Send trial point for checks
+        self._ready_for_tell = True
+        return np.array(self._proposed, copy=True)
+
+
+    def tell(self, reply):
+        """ See :meth:`pints.SingleChainMCMC.tell()`. """
+
+        # Check ask/tell pattern
+        if not self._ready_for_tell:
+            raise RuntimeError('Tell called before proposal was set.')
+        self._ready_for_tell = False
+
+        # Unpack reply
+        fx = np.asarray(reply, dtype=float)
+
+        # Very first call
+        if self._current is None:
+
+            # Check first point is somewhere sensible
+            if not np.isfinite(fx):
+                raise ValueError(
+                    'Initial point for MCMC must have finite logpdf.')
+
+            # Set current sample, log pdf of current sample and initialise proposed
+            # sample for next iteration
+            self._current = np.array(self._x0, copy=True)
+            self._current_log_pdf = fx
+            self._proposed = np.array(self._current, copy=True)
+
+            # Sample height of the slice log_y for next iteration
+            self._e = np.random.exponential(1)
+            self._current_log_y = self._current_log_pdf - self._e
+
+            # Increase number of samples found
+            self._mcmc_iteration += 1
+
+            # Set flag to true as we need to initialise the interval expansion for
+            # next iteration
+            self._first_expansion = True
+
+            # Return first point in chain, which is x0
+            return np.array(self._current, copy=True)
+
+        # While we expand the interval ``I=(l,r)``, we return None
+        if self._interval_found == False:
+
+            # Set the log_pdf of the interval edge that we are expanding
+            if self._set_l:
+                self._fx_l = fx
+            elif self._set_r:
+                self._fx_r = fx
+            elif self._init_left:
+                self._fx_l = fx
+                self._init_left = False
+            elif self._init_right:
+                self._fx_r = fx
+                self._init_right = False
+
+            return None
+
+        # Do ``Threshold Check`` to check if the proposed point is within the slice
+        if self._current_log_y < fx:
+
+            # Reset flag to true as we need to initialise the interval for the update of the
+            # next parameter
+            self._first_expansion = True
+
+            # Reset flag to false since we still need to estimate the interval for the next parameter
+            self._interval_found = False
+
+
+            # If we have updated all the parameters of the sample, start constructing next sample
```

Comment:
This comment confused me as thought, "Surely we start constructing the sample in Ask()''. Think we can just say "Reset active parameter indices".